.github/workflows/ci.yml

```
@@ -119,6 +135,62 @@ jobs:
         flags: unittests
         name: codecov-umbrella
 
+  unit-test-modules:
+    name: Unit Tests (Modules)
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: crypto
+            make1: no-op
+            make2: test
+            retries: 1
+            race: 1
+          - name: insecure
+            make1: install-tools
+            make2: test
+            retries: 3
+            race: 1
+          - name: integration
+            make1: install-tools
+            make2: test
+            retries: 3
+            race: 0
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache: true
+      - name: Run tests (${{ matrix.name }})
+        if: github.actor != 'bors[bot]'
+        env:
+          RACE_DETECTOR: ${{ matrix.race }}
+        # run `make1` target before running `make2` target inside each module's root
+        run: |
+          make ${{ matrix.make1 }}
+          cd ${{ matrix.name }}
+          make ${{ matrix.make2 }}          
```

These 2 lines could be replaced by:
```suggestion
make -C ${{ matrix.name }} ${{ matrix.make2 }}
```

---

Good idea - updated in 406e9f80be1eb8b3d1880233b0488c72074cf0b2